File: CountWordFrequency.js
let wordArray = ["Apple", "Banana", "Orange", "Apple", "Orange", "Banana", "Apple"]

let wordFrequencyCount = wordArr => {
    let wordMap = new Map()
    wordArr.forEach(word => {
        if(!wordMap.has(word)){
            wordMap.set(word, 1)
        }else{
            wordMap.set(word, wordMap.get(word) + 1)
        }
    });
    return wordMap
}

let result = wordFrequencyCount(wordArray)
console.log(result)

File: CountWordFrequency.java
import java.util.HashMap;

class CountWordFrequency {

    public static void main(String[] args) {
        System.out.println(wordFrequencyCount(new String[]{"Apple", "Banana", "Orange", "Apple", "Orange", "Banana", "Apple"}));
    }

    private static HashMap<String, Integer> wordFrequencyCount(String[] wordArray) {
        HashMap<String, Integer> map = new HashMap<>();
        for (String word : wordArray) {
            if(!map.containsKey(word)){
                map.put(word, 1);
                System.out.println("Word: " + word + ", " + "map: " + map);
            }else{
                map.put(word, map.get(word) + 1);
                System.out.println("Word: " + word + " " + "map: " + map);
            }
            
        }

        return map;
    }
}